swarm/api/client/client.go

```diff
@@ -175,7 +175,10 @@ func (c *Client) UploadDirectory(dir, defaultPath, manifest string, toEncrypt bo
 	} else if !stat.IsDir() {
 		return "", fmt.Errorf("not a directory: %s", dir)
 	}
-	return c.TarUpload(manifest, &DirectoryUploader{dir, defaultPath}, toEncrypt)
+	if _, err := os.Stat(filepath.Join(dir, defaultPath)); err != nil {
```

Comment:
Should this only be done if `defaultPath` is set?

Reply:
Thanks, that is correct.